src/Nest/Document/Multiple/Bulk/BulkRequest.cs

```diff
@@ -10,17 +10,23 @@ namespace Nest
 	public partial interface IBulkRequest
 	{
 		[IgnoreDataMember]
-		BulkOperationsCollection<IBulkOperation> Operations { get; set; }
+		IList<IBulkOperation> Operations { get; set; }
 	}
 
 	public partial class BulkRequest
 	{
-		public BulkOperationsCollection<IBulkOperation> Operations { get; set; }
+		public IList<IBulkOperation> Operations { get; set; }
 	}
 
 	public partial class BulkDescriptor
 	{
-		BulkOperationsCollection<IBulkOperation> IBulkRequest.Operations { get; set; } = new BulkOperationsCollection<IBulkOperation>();
+		private readonly BulkOperationsCollection<IBulkOperation> _operations = new BulkOperationsCollection<IBulkOperation>();
+
+		IList<IBulkOperation> IBulkRequest.Operations
+		{
+			get => _operations;
+			set { }
```

Comment:
I think we should either:

1. `throw new InvalidOperationException()` here, indicating that it cannot be set.

or

2. make _operations an `IList<IBulkOperation>`, initialize a new instance of `new BulkOperationsCollection<IBulkOperation>()` on the getter if the backing _operations is null, and allow someone to set _operations with the setter

Reply:
We should! Will go for option 1.